### ISSUE TITLE
Error when trying to setup master/slave with key

### ISSUE THREAD

**Issue by frock81** [bertvv/ansible-role-bind#114]
I'm trying to setup a master/slave scenario in conjunction with a dhcp shared secret key to provide dynamic dns updates.

I'm getting the following errors:

```
TASK [bertvv.bind : Master | Main BIND config file (master)] *******************
fatal: [master.localdomain]: FAILED! => {"changed": false, "checksum": "1e3b5464474791dfafe1e054ab38d10555b03624", "exit_status": 1, "msg": "failed to validate", "stderr": "", "stderr_lines": [], "stdout": "/hom
e/vagrant/.ansible/tmp/ansible-tmp-1588965959.41-9257-122598536422986/source:58: open: /etc/bind//auth_transfer.conf: file not found\n", "stdout_lines": ["/home/vagrant/.ansible/tmp/ansible-tmp-1588965959.41-925
7-122598536422986/source:58: open: /etc/bind//auth_transfer.conf: file not found"]}

TASK [bertvv.bind : Set up the machine as a slave DNS server] ******************
included: /etc/ansible/roles/bertvv.bind/tasks/slave.yml for slave.localdomain

TASK [bertvv.bind : Slave | Main BIND config file (slave)] *********************
fatal: [slave.localdomain]: FAILED! => {"changed": false, "checksum": "6fce3294c6909cf44db814973865b2380d46be86", "exit_status": 1, "msg": "failed to validate", "stderr": "", "stderr_lines": [], "stdout": "/home
/vagrant/.ansible/tmp/ansible-tmp-1588965960.21-9275-228126818034044/source:57: open: /etc/bind//auth_transfer.conf: file not found\n", "stdout_lines": ["/home/vagrant/.ansible/tmp/ansible-tmp-1588965960.21-9275
-228126818034044/source:57: open: /etc/bind//auth_transfer.conf: file not found"]}
```

I get the same error from docker-tests when modifying the file `test.yml` to include the `bind_dns_keys` and `bind_extra_include_files`. 

I'm not sure if I am doing something wrong.

Debugging the role seems to me that the auth transfer template task should come before the master and slave setup tasks in the file _tasks/main.yml_.

```
- name: Set up the machine as a master DNS server
  include_tasks: master.yml
  when: bind_zone_master_server_ip in ansible_all_ipv4_addresses

- name: Set up the machine as a slave DNS server
  include_tasks: slave.yml
  when: bind_zone_master_server_ip not in ansible_all_ipv4_addresses

# file to set keys for XFR authentication
- name: create extra config file for authenticated XFR request
  tags: pretask
  template:
    src: auth_transfer.j2
    dest: "{{ bind_conf_dir }}/{{ auth_file }}"
    mode: 0640
    owner: root
    group: "{{ bind_group }}"
  when: bind_dns_keys is defined and bind_dns_keys|length > 0
```

When the template task is put before the master and slave tasks, the docker-tests pass.

I created a vagrant environment to reproduce the error in: https://github.com/frock81/ms-key-error

In there I added the following values to the vars to get the error.

```
    bind_dns_keys:
      - name: DHCP_UPDATER
        algorithm: hmac-md5
        secret: "TRnWB2o0SQ1VQasL+gNOdg=="    # insecure key
    bind_extra_include_files:
      - "{{ bind_auth_file }}"
```

**Comment by frock81** [comment]
Added https://github.com/bertvv/ansible-role-bind/pull/115 as a proposal to fix the issue.

**Comment by frock81** [comment]
Besides that, not a issue but related to the auth transfer, the definition of the variable `bind_conf_dir: "/etc/named/"` for Redhat and `bind_conf_dir: "/etc/bind/"` for Debian with a trailing slash makes added to the definition of `bind_auth_file: "{{ bind_conf_dir }}/{{ auth_file }}"` for both families causes a double slash while templating _named.conf_.

It does not generate an error but can be easily avoided by removing the trailing slash from `bind_conf_dir`.

**Comment by bertvv** [comment]
I merged #115, thanks!
